firebase-database/src/main/java/com/google/firebase/database/connection/PersistentConnectionImpl.java

```diff
@@ -761,17 +761,11 @@ private void tryScheduleReconnect() {
                 .addOnSuccessListener(
                     executorService,
                     aVoid -> {
-                      // Someone could have interrupted us while fetching the token,
-                      // marking the connection as Disconnected
-                      if (connectionState == ConnectionState.GettingToken) {
-                        if (thisGetTokenAttempt == currentGetTokenAttempt) {
+                      if (thisGetTokenAttempt == currentGetTokenAttempt) {
```

Comment:
This aligns the code back with the pre-AppCheck flow: https://github.com/firebase/firebase-android-sdk/commit/6da8617afe903126afac46ef71b1d857dcfe7ef9#diff-ff27763e8a3b1c249d9220a66053d6ca0dcdd3314480a2eb756a07d7fdf046c9L720